test/reduce-e2e/reduce_test.go

```diff
@@ -84,30 +100,42 @@ func (r *ReduceSuite) TestSimpleNonKeyedReducePipeline() {
 	// port forward source vertex(to publish messages)
 	defer w.VertexPodPortForward("in", 8443, dfv1.VertexHTTPSPort).
 		TerminateAllPodPortForwards()
-
-	// publish messages to source vertex, with event time starting from 60000
-	startTime := 60000
-	for i := 0; i < 300; i++ {
-		eventTime := strconv.Itoa(startTime + i*1000)
-
-		HTTPExpect(r.T(), "https://localhost:8443").POST("/vertices/in").WithBytes([]byte("1")).WithHeader("X-Numaflow-Event-Time", eventTime).
-			Expect().
-			Status(204)
-		HTTPExpect(r.T(), "https://localhost:8443").POST("/vertices/in").WithBytes([]byte("2")).WithHeader("X-Numaflow-Event-Time", eventTime).
-			Expect().
-			Status(204)
-		HTTPExpect(r.T(), "https://localhost:8443").POST("/vertices/in").WithBytes([]byte("3")).WithHeader("X-Numaflow-Event-Time", eventTime).
-			Expect().
-			Status(204)
-	}
+	done := make(chan struct{})
+	go func() {
+		// publish messages to source vertex, with event time starting from 60000
+		startTime := 60000
+		for i := 0; true; i++ {
+			select {
+			case <-ctx.Done():
```

Comment:
if this `context.Done` happens, it means the test has to be marked as a failure right?

Reply:
yeah I have missed that